tamr_unify_client/models/dataset/resource.py

```diff
@@ -94,6 +94,188 @@ def status(self) -> DatasetStatus:
+        geo_attr_names = {
+            "point",
+            "multiPoint",
+            "lineString",
+            "multiLineString",
+            "polygon",
+            "multiPolygon",
+        }
+        geo_attrs = [
+            attr.name
+            for attr in self.attributes
+            if "RECORD" == attr.type.base_type
+            and geo_attr_names.intersection(
+                {sub_attr.name for sub_attr in attr.type.attributes}
+            )
+        ]
+        # We select the first such attribute as the geometry
+        if geo_attrs:
+            geo_attr = geo_attrs[0]
+        else:
+            geo_attr = None
+        return geo_attr
+
+    @staticmethod
+    def _record_to_feature(record, key_value, key_attrs, geo_attr):
+        """Convert a Unify record to a Python Geo Interface Feature
+
+        :param record: Unify record
+        :param key_value: Function to extract the value of the primary key from the record
+        :param key_attrs: Set of attributes that comprise the primary key for the record
+        :param geo_attr: The singular attribute to use as the geometry
+        :return: map from str to object
+        """
+        feature = {"type": "Feature", "id": key_value(record)}
+        reserved = {"bbox", geo_attr}.union(key_attrs)
+        conversion = {
+            "point": "Point",
+            "multiPoint": "MultiPoint",
+            "lineString": "LineString",
+            "multiLineString": "MultiLineString",
+            "polygon": "Polygon",
+            "multiPolygon": "MultiPolygon",
+        }
+        if geo_attr and geo_attr in record:
+            src_geo = record[geo_attr]
+            for unify_attr in conversion.keys():
+                if unify_attr in src_geo and src_geo[unify_attr]:
+                    feature["geometry"] = {
+                        "type": conversion[unify_attr],
+                        "coordinates": src_geo[unify_attr],
+                    }
+                    break
+        if "bbox" in record:
+            feature["bbox"] = record["bbox"]
+        non_reserved = set(record.keys()).difference(reserved)
+        if non_reserved:
+            feature["properties"] = {attr: record[attr] for attr in non_reserved}
+        return feature
+
+    @staticmethod
+    def _feature_to_record(feature, key_attrs, geo_attr):
+        """Convert a Python Geo Interface Feature to a Unify record
+
+        :param feature: Python Geo Interface Feature
+        :param key_attrs: Sequence of attributes that comprise the primary key for the record
+        :param geo_attr: The singluar attribute on the record to use for the geometry
+        :return: dict
+        """
+        if hasattr(feature, "__geo_interface__"):
+            feature = feature.__geo_interface__
+
+        record = {}
+
+        props = feature.get("properties")
+        if props:
+            for prop in props:
+                record[prop] = props[prop]
+
+        geometry = feature.get("geometry")
```

Comment:
Can a feature exist without a geometry and/or bbox? I feel like there should be an assert somewhere to confirm this is a valid geometry.

Reply:
Sadly, both geometry and bbox are optional in a feature.
A FeatureCollection with neither is an ordinary dataset.